From: Russ Allbery <eagle@eyrie.org>
Date: Fri, 28 Feb 2014 00:41:47 -0800
Subject: Specify test databases in kdc.conf

Setting the database when calling kadm5_init_with_password_ctx
doesn't actually work.  Set all the database paths in kdc.conf.
---
 t/data/kdb/kdc.conf | 6 +++++-
 t/kadmin/heimdal.t  | 7 ++-----
 2 files changed, 7 insertions(+), 6 deletions(-)

diff --git a/t/data/kdb/kdc.conf b/t/data/kdb/kdc.conf
index a39411e..c01ab33 100644
--- a/t/data/kdb/kdc.conf
+++ b/t/data/kdb/kdc.conf
@@ -3,6 +3,10 @@
 
 [kdc]
     database = {
-        dbname    = db:./heimdal
+        dbname    = db:./t/tmp/heimdal
         realm     = TEST.EXAMPLE.COM
     }
+    database = {
+        dbname    = db:./t/tmp/bogus
+        realm     = BOGUS.EXAMPLE.COM
+    }
diff --git a/t/kadmin/heimdal.t b/t/kadmin/heimdal.t
index 77af47d..7f25c2f 100755
--- a/t/kadmin/heimdal.t
+++ b/t/kadmin/heimdal.t
@@ -49,14 +49,12 @@ END {
     rmdir('t/tmp');
 }
 
-# Force use of our local kdc.conf, since kadmin library initialization wants
-# to know the host to realm mapping for some reason.
+# Force use of our local kdc.conf.
 local $ENV{KRB5_CONFIG} = 't/data/kdb/kdc.conf';
 
 # Create the Authen::Kerberos::Kadmin object.
 my $kadmin = Authen::Kerberos::Kadmin->new(
     {
-        db_name => 'db:./t/tmp/heimdal',
         realm   => 'TEST.EXAMPLE.COM',
         server  => 1,
     }
@@ -72,8 +70,7 @@ is($@, q{}, '...with no exception');
 # The same should fail if we attempt it with an unknown database.
 $kadmin = Authen::Kerberos::Kadmin->new(
     {
-        db_name => 'db:./t/tmp/bogus',
-        realm   => 'TEST.EXAMPLE.COM',
+        realm   => 'BOGUS.EXAMPLE.COM',
         server  => 1,
     }
 );
